### PR TITLE
ENH: Improve PR workflows

### PR DESCRIPTION
**Pull request by ant0nsc**
* Run `pytest` and `pytest` on GPUs in parallel rather than sequentially.
* Cancel existing runs from previous pushes
* Fix dependency problem in hi-ml PR
* Reduce logging noise from `numba`

### CODE REVIEW

**Comment by ant0nsc** [comment]
closes #698 #699 #700

File: .github/actions/cancel_azureml_jobs/cancel_aml_jobs.py [hunk 1]
```diff
@@ -36,7 +36,10 @@ def cancel_running_and_queued_jobs() -> None:
             print(f"Skipping {status_suffix}")
         else:
             print(f"Cancelling {status_suffix}")
-            run.cancel()
+            try:
+                run.cancel()
+            except Exception as ex:
+                print(f"Unable to cancel {status_suffix}: {ex}")
```
Comment (kenza-bouzid) [inline .github/actions/cancel_azureml_jobs/cancel_aml_jobs.py:42]: What can cause this?

File: hi-ml-cpath/.vscode/launch.json [hunk 2]
```diff
@@ -85,5 +85,21 @@
...
             "justMyCode": false,
         },
+        {
+            "name": "Upload heatmap to DSA",
+            "type": "python",
+            "request": "launch",
+            "program": "${workspaceFolder}/src/health_cpath/utils/girder.py",
+            "args": [
+                "--dsa-url=https://dsa.westeurope.cloudapp.azure.com/",
+                "--dsa-key=FFw1x8bnzG14Fv88dq3Hh4liFfZFxQgL94smESfB",
```
Comment (kenza-bouzid) [inline hi-ml-cpath/.vscode/launch.json:95]: is it okay to have this public?

File: hi-ml-cpath/Makefile [hunk 3]
```diff
@@ -81,6 +81,11 @@ pytest:
 pytest_coverage:
 	pytest --cov=health_cpath --cov SSL --cov-branch --cov-report=html --cov-report=xml --cov-report=term-missing --cov-config=.coveragerc
 
+# run pytest using GPUs via an AzureML job
+pytest_gpu:
+	python ../hi-ml-azure/run_pytest.py --mark=gpu --cluster=pr-gpu --conda_env=environment.yml --folder=hi-ml-cpath --coverage_module=health_cpath
```
Comment (kenza-bouzid) [inline hi-ml-cpath/Makefile:86]: you might want to add --strictly_aml_v1=True here. (it's true by default but the docu is misleading, I updated it in #695)

**Review by kenza-bouzid — APPROVED** [review]
Looking good!

File: hi-ml-cpath/.vscode/launch.json (hunk 2 above)
Comment (ant0nsc) [reply hi-ml-cpath/.vscode/launch.json:95]: nope. fixing.

**Review by peterhessey — APPROVED** [review]
LGTM :)

**Pull request merged**